lib/ecto/changeset.ex

```diff
@@ -1570,4 +1570,61 @@ defmodule Ecto.Changeset do
     model.__schema__(:association, assoc) ||
       raise(ArgumentError, "cannot add constraint to model because association `#{assoc}` does not exist")
   end
+
+  @doc """
+  Traverses changeset errors and applies function to error messages.
+
+  A changeset is supplied along with a function to apply to each
+  error message as the changeset is traversed. The error message
+  function receives a single arugment matching either:
+
+    * `{message, count}` - The string error message and count,
+      for example `"should be at least %{count} characters"`
+    * `message` - The string error message
```

Comment:
Should we impose this format when the errors in the changeset is `{String.t, Keyword.t}`. This will be backwards incompatible if we have multiple variables to interpolate.